app/gui2/src/components/GraphEditor/widgets/WidgetText.vue

```diff
@@ -1,23 +1,46 @@
 <script setup lang="ts">
-import EnsoTextInputWidget from '@/components/widgets/EnsoTextInputWidget.vue'
+import NodeWidget from '@/components/GraphEditor/NodeWidget.vue'
+import AutoSizedInput from '@/components/widgets/AutoSizedInput.vue'
 import { Score, WidgetInput, defineWidget, widgetProps } from '@/providers/widgetRegistry'
+import { useGraphStore } from '@/stores/graph'
 import { Ast } from '@/util/ast'
-import type { TokenId } from '@/util/ast/abstract'
-import { asNot } from '@/util/data/types'
+import { MutableModule } from '@/util/ast/abstract'
 import { computed } from 'vue'
 
 const props = defineProps(widgetProps(widgetDefinition))
-const value = computed({
+const graph = useGraphStore()
+
+const inputTextLiteral = computed((): Ast.TextLiteral | undefined => {
+  if (props.input instanceof Ast.TextLiteral) return props.input
```

Comment:
WidgetInput type cannot be Ast.TextLiteral. Do you mean props.input.value?

Reply:
Yeah. It seems it worked and passed my testing only because the "Ast.parse" fallback below.